### PR TITLE
CFY-5317 Pre-upgrade validation tests.

### PR DESCRIPTION
**Pull request by tehasdf**

### CODE REVIEW

**Comment by tehasdf** [comment]
Requires #434 for the assertions and cloudify-cosmo/cloudify-manager-blueprints#345 to pass.

File: cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+        with self.cfy.maintenance_mode():
+            try:
+                self.cfy.upgrade_manager(
+                    blueprint_path=blueprint_path,
+                    inputs_file=inputs,
+                    validate_only=True)
+            except sh.ErrorReturnCode as e:
+                self.assertIn('rabbitmq properties must not change',
+                              e.stdout.lower())
```
Comment (adamlavie) [inline cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py]: According to the code, we can validate all 5 immutable properties (rabbitmq_username, rabbitmq_pasword, rabbitmq_endpoint_ip, rabbitmq_ssl_enabled, broker_cert_path) with one iteration. I say why not while we are at it.

File: cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+                                  'succeeded, but it should have failed - '
+                                  'node properties and rollback directories '
+                                  'are missing'.format(node_name))
+
+    def test_rabbit_credentials_changed(self):
+        """Validation doesn't allow changing rabbitmq credentials.
+
+        Try upgrading with a changed rabbitmq password, check that validation
+        breaks.
+        """
```
Comment (adamlavie) [inline cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py]: Test immutable properties in the manager as well (ssl, security, ssh_user)

File: cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        fetched_properties.seek(0)
+        return json.load(fetched_properties)
+
+    def put_node_properties(self, node_name, properties):
+        """Overwrite node properties in manager's storage"""
+        properties_path = self.get_node_properties_path(node_name)
+        with self.manager_env_fabric() as fabric:
+            fabric.put(StringIO(json.dumps(properties)), properties_path)
+
+    def test_default_validations(self):
```
Comment (adamlavie) [inline cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py]: Have the test functions at the top of the class for readability

File: cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py (hunk 1 above)
Comment (tehasdf) [inline cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py]: Alright, now getting all these inputs from `.get_changed_rabbit_properties()`

File: cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py (hunk 2 above)
Comment (tehasdf) [inline cosmo_tester/test_suites/test_manager_upgrade/preupgrade_validations_test.py]: Done in `test_manager_security_settings_changed` and `test_manager_ssh_user_changed`

**Pull request merged**